poll.go

```diff
@@ -152,14 +154,33 @@ func (h *Headscale) PollNetMapHandler(ctx *gin.Context) {
 		Str("id", ctx.Param("id")).
 		Str("machine", machine.Name).
 		Msg("Loading or creating update channel")
-	updateChan := make(chan struct{})
 
-	pollDataChan := make(chan []byte)
+	// TODO: could probably remove all that duplication once generics land.
```

Comment:
\o/ generics!

Reply:
To be honest, I kind of forgot about this comment. That block of code had quite a bit more duplication when I wrote it, and I'm not in favor of using generics just to drop a mere three lines of code.

Reply:
Yes, this isnt the only place were Generics will help us :P